Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,522 @@
+
+git_repository(
+    name = "build_bazel_rules_apple",
+    remote = "https://github.com/bazelbuild/rules_apple.git",
+    commit = "[SOME_HASH_VALUE]",
+)
+```
+
+The first line of code calls the function, `load`. The load function imports
+symbols from a `.bzl` file.  In Objective-C or Swift, it's like importing a
+header file.
+
+```
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+```
+_load `git_repository` from the internal [bazel tools](https://github.com/bazelbuild/bazel/issues/4301) repository_
+
+The next line of code calls `git_repository` and defines the repository
+`build_bazel_rules_apple` from the git repository,
+`https://github.com/bazelbuild/rules_apple` for a given `commit`. 
+
+### Adding an iOS application with BUILD files
+
+`BUILD` files are where all the targets are defined. For Apple developers, this
+often includes apps, tests, app extensions, frameworks, and libraries.
+
+First, let's walk through creating a basic iOS application. In the root of the
+project, let's create the `BUILD` file. First, a library for the iOS
+application sources. The following code defines an `objc_library`, `sources`.
+
+```
+# /path/to/myproject/BUILD
+objc_library(
+    name="sources",
+    srcs=["main.m"]
+)
+```
+
+Next, create the application target with `rules_apple`'s `ios_application` rule.
+```
+# /path/to/myproject/BUILD
+...
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
+ios_application(
+    name = "ios-app",
+    bundle_id = "com.bazel-bootcamp.some",
+    families = ["iphone"],
+    infoplists = ["Info.plist"],
+    minimum_os_version = "[MINIMUM VERSION]",
+    # Add `sources` as a dependency
+    deps = [ ":sources" ],
+)
+```
+
+In Xcode this is similar to navigating in the GUI and hitting `File -> New
+Target`. Together, the `ios_application` and `objc_library` targets together
+would be represented in Xcode as:
+
+![Docs](XcodeExampleOfiOSProject.png)
+	
+### Bazel command line
+
+In Xcode, command line builds are achieved through the command line interface
+`xcodebuild`.  to build the scheme `ios-app` from `MyProject.xcworkspace`,
+`xcodebuild`, would be invoked as:
+
+```
+xcodebuild -workspace MyProject.xcworkspace -scheme ios-app
+```
+
+Bazel exposes a command line which can also do builds. In Bazel, _every_ target
+is in the global WORKSPACE and there is no notion of schemes.
+
+```
+bazel build ios-app
+```
+
+The Bazel command line has hundreds of options, which can be found in the
+[Bazel
+documentation](https://docs.bazel.build/versions/master/command-line-reference.html).
+For iOS developers, a set of useful flags is available at
+[bazel-ios-users](https://github.com/ios-bazel-users/ios-bazel-users/blob/master/UsefulFlags.md).
+
+### Configurable build attributes
+
+In addition to the default Bazel options, it's common to create custom
+configuration settings to customize builds. Together, `select` and
+`config_setting` yield configurable build attributes.
```

Comment:
Add clarification that "custom configuration" could mean something like the difference between a build for app store vs a build for development.

Reply:
Good suggestion, updated 👍